tests/rptest/tests/strict_data_init_test.py

```diff
@@ -0,0 +1,43 @@
+import re
+
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.services.redpanda import RedpandaService, RESTART_LOG_ALLOW_LIST
+from rptest.clients.rpk import RpkTool
+from rptest.services.cluster import cluster
+from rptest.services.utils import BadLogLines
+
+STRICT_DATA_LOG_ALLOW_LIST = RESTART_LOG_ALLOW_LIST + [
+    re.compile("not found, is the expected filesystem mounted?"),
+    re.compile(".redpanda_data_dir does not exist")
+]
+
+
+class StrictDataInitTest(RedpandaTest):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, extra_rp_conf={}, **kwargs)
+        self.rpk = RpkTool(self.redpanda)
+
+    @cluster(num_nodes=1, log_allow_list=STRICT_DATA_LOG_ALLOW_LIST)
+    def test_strict_data_init_enabled(self):
+        target_node = self.redpanda.nodes[0]
+
+        # Enable the `storage_strict_data_init` property
+        # the node should fail to restart with it enabled.
+        self.rpk.cluster_config_set("storage_strict_data_init", "true")
+        self.redpanda.stop_node(target_node)
+        self.redpanda.start_node(target_node, expect_fail=True)
+
+        # Verify the reason for the node not starting is what
+        # we expect it to be.
+        try:
+            self.redpanda.raise_on_bad_logs()
+        except BadLogLines as b:
+            bad_lines = b.node_to_lines[target_node]
+            assert any(".redpanda_data_dir does not exist" in b
```

Comment:
this string is duplicated: should define the string once up top, then use it both here and in your allow list

Reply:
Got it, will define it only once.